### PR TITLE
cleanup: remove non-trivially destructible static variables

### PR DESCRIPTION
**Pull request by dbolduc**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6995)

### CODE REVIEW

**Review by devjgm — APPROVED** [review]
Thanks for the cleanup!

File: google/cloud/bigtable/benchmarks/random_mutation.cc [hunk 1]
```diff
@@ -27,9 +27,9 @@ bigtable::Mutation MakeRandomMutation(google::cloud::internal::DefaultPRNG& gen,
 }
 
 std::string MakeRandomValue(google::cloud::internal::DefaultPRNG& generator) {
-  static std::string const kLetters(
+  static std::string const* const kLetters = new std::string(
```
Comment (devbww) [inline google/cloud/bigtable/benchmarks/random_mutation.cc:30]: Opt: Some suggest avoiding the `std::string` repetition with ...
```
static auto const* = new std::string(...);
```

**Review by coryan — APPROVED** [review]
so many starts

File: google/cloud/internal/build_info.cc.in [hunk 2]
```diff
@@ -29,18 +29,19 @@ std::string compiler() {
 
 // NOLINTNEXTLINE(readability-identifier-naming)
 std::string compiler_flags() {
-  static char const kCompilerFlags[] =
-      R"""(@CMAKE_CXX_FLAGS@ ${CMAKE_CXX_FLAGS_${GOOGLE_CLOUD_CPP_BUILD_TYPE_UPPER}})""";
-  return kCompilerFlags;
+  static auto const* const kCompilerFlags = new auto(
+      R"""(@CMAKE_CXX_FLAGS@ ${CMAKE_CXX_FLAGS_${GOOGLE_CLOUD_CPP_BUILD_TYPE_UPPER}})""");
+  return *kCompilerFlags;
```
Comment (devbww) [inline google/cloud/internal/build_info.cc.in:34]: `char[]` has a trivial destructor, so this didn't really buy you anything (but it probably cost you call-time synchronization and a dynamic allocation).

That said we could make the function static a `std::string`, but whether it is worth it depends on the call pattern.

**Review by dbolduc — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6995)* status: 0 of 11 files reviewed, 2 unresolved discussions (waiting on @devbww)

___
*[google/cloud/bigtable/benchmarks/random_mutation.cc, line 30 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6995#-Meg5M9C0sD_Xky3JfLf-r1-30:-MejdXhq5RuZHcl75qnP:b-kukirg) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/b3bfc3d68b5e010e5d160840ffb5891c75d0f87e/google/cloud/bigtable/benchmarks/random_mutation.cc#L30)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Opt: Some suggest avoiding the `std::string` repetition with ...
```
static auto const* = new std::string(...);
```
</blockquote></details>

True. I think this makes the code 0.1% nicer. Fixed.
___
*[google/cloud/internal/build_info.cc.in, line 34 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6995#-Meg5M9C0sD_Xky3JfLi-r1-34:-Meje0PJ8_a5PEMZ8oYh:bp3aow6) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/b3bfc3d68b5e010e5d160840ffb5891c75d0f87e/google/cloud/internal/build_info.cc.in#L34)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

`char[]` has a trivial destructor, so this didn't really buy you anything (but it probably cost you call-time synchronization and a dynamic allocation).

That said we could make the function static a `std::string`, but whether it is worth it depends on the call pattern.
</blockquote></details>

Reverted.

**Pull request merged**